crates/dbs-utils/src/epoll_manager.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A simple wrapper over event_manager::EventManager to solve possible deadlock.
+
+use anyhow::{anyhow, Result};
+use std::sync::{Arc, Mutex};
+
+pub use event_manager::{
+    Error, EventManager, EventOps, EventSet, Events, MutEventSubscriber, RemoteEndpoint,
+    SubscriberId, SubscriberOps,
+};
+
+/// Type of epoll subscriber.
+pub type EpollSubscriber = Box<dyn MutEventSubscriber + Send>;
+
+type EpollManagerImpl = Arc<Mutex<EventManager<EpollSubscriber>>>;
+
+/// A wrapper struct over EventManager to solve possible deadlock.
+///
+/// It's a rather tough topic to deal with the epoll event manager in rust way.
+/// The event_manager::EventManager is designed for single-threaded environment and it leaves
+/// the task for concurrent access to the clients.
+/// There are two types of threads involved, epoll worker thread and vCPU threads.
+/// To reduce overhead, the epoll worker thread calls epoll::wait() without timeout, so the
+/// worker thread will hold the EpollManagerImpl::Mutex for undetermined periods. When the vCPU
+/// threads tries to activate virtio devices, they need to acquire the same EpollManagerImpl::Mutex.
+/// Thus the vCPU threads may block for an undetermined time. To solve this issue, we perform
+/// an kick()/try_lock() loop to wake up the epoll worker thread from sleeping.
+#[derive(Clone)]
+pub struct EpollManager {
+    pub mgr: EpollManagerImpl,
+    endpoint: Arc<Mutex<RemoteEndpoint<EpollSubscriber>>>,
+}
+
+impl EpollManager {
+    /// Add a new epoll event subscriber.
+    pub fn add_subscriber(&self, handler: EpollSubscriber) -> SubscriberId {
+        let _ = self.endpoint.lock().unwrap().kick();
```

Comment:
is the let_ part necessary？

Reply:
`kick` returns Result, here we want to ignore the error.